scripts/validate_changelog.py

```diff
@@ -0,0 +1,172 @@
+    return bool(match)
+
+
+def is_module_or_plugin(ref: str) -> bool:
+    prefix_list = (
+        "plugins/modules",
+        "plugins/action",
+        "plugins/inventory",
+        "plugins/lookup",
+        "plugins/filter",
+        "plugins/connection",
+        "plugins/become",
+        "plugins/cache",
+        "plugins/callback",
+        "plugins/cliconf",
+        "plugins/httpapi",
+        "plugins/netconf",
+        "plugins/shell",
+        "plugins/strategy",
+        "plugins/terminal",
+        "plugins/test",
+        "plugins/vars",
+    )
+    return ref.startswith(prefix_list)
+
+
+def is_documentation_file(ref: str) -> bool:
+    prefix_list = (
+        "docs/",
+        "plugins/doc_fragments",
+    )
+    return ref.startswith(prefix_list)
+
+
+def should_skip_changelog(changes: Dict[Any, Any]) -> bool:
+    # Validate Pull request add new modules and plugins
+    if any(is_module_or_plugin(x) for x in changes["A"]):
+        return True
+
+    # Validate documentation changes only
+    all_files = changes["A"] + changes["M"] + changes["D"]
+    if all(is_documentation_file(x) for x in all_files):
+        return True
+
+    return False
+
+
+def is_valid_changelog_format(path: str) -> bool:
+    try:
+        # https://github.com/ansible-community/antsibull-changelog/blob/main/docs/changelogs.rst#changelog-fragment-categories
+        changes_type = (
+            "release_summary",
+            "breaking_changes",
+            "major_changes",
+            "minor_changes",
+            "removed_features",
+            "deprecated_features",
+            "security_fixes",
+            "bugfixes",
+            "known_issues",
+            "trivial",
+        )
+        with open(path, "rb") as file_desc:
+            result = list(yaml.safe_load_all(file_desc))
+
+        for section in result:
+            for key in section.keys():
+                if key not in changes_type:
+                    msg = f"{key} from {path} is not a valid changelog type"
+                    logger.error(msg)
+                    return False
+                if not isinstance(section[key], list):
+                    logger.error(
+                        "Changelog section %s from file %s must be a list,"
+                        " '%s' found instead.",
+                        key,
+                        path,
+                        type(section[key]),
+                    )
+                    return False
+        return True
+    except (IOError, yaml.YAMLError) as exc:
+        msg = f"yaml loading error for file {path} -> {exc}"
+        logger.error(msg)
+        return False
+
+
+def run_command(cmd: str) -> Tuple[Union[int, Any], str, str]:
+    with subprocess.Popen(
+        cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE, shell=True
+    ) as proc:
+        out, err = proc.communicate()
+        return proc.returncode, out.decode("utf-8"), err.decode("utf-8")
```

Comment:
If we're going to assume encoding anyway, why not let subprocess convert it for us?
```suggestion
        cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE, shell=True, encoding="utf-8",
    ) as proc:
        out, err = proc.communicate()
        return proc.returncode, out, err
```